pkg/terminal/p400/reader_methods.go

```diff
@@ -0,0 +1,397 @@
+type PaymentMethod struct {
+}
+
+// ReaderActivateContent represents the shape of the serialized protobuf sent to Rabbit Service for activating a terminal session
+type ReaderActivateContent struct {
+	POSActivationToken string          `json:"pos_activation_token"`
+	StoreName          string          `json:"store_name"`
+	POSDeviceID        string          `json:"pos_device_id"`
+	POSSoftwareInfo    POSSoftwareInfo `json:"pos_software_info"`
+}
+
+// ReaderActivateResponse is the RPC response from calling the activateTerminal method
+type ReaderActivateResponse struct {
+	SessionToken string `json:"session_token"`
+}
+
+// LineItem belongs to the Cart protobuf shape for updating the reader display via Rabbit Service
+type LineItem struct {
+	Description string `json:"description"`
+	Amount      int    `json:"amount"`
+	Quantity    int    `json:"quantity"`
+}
+
+// Cart belongs to the ReaderDisplayContent protobuf sent to Rabbit Service that updates the reader display
+type Cart struct {
+	LineItems []LineItem `json:"line_items"`
+	Tax       int        `json:"tax"`
+	Total     int        `json:"total"`
+	Currency  string     `json:"currency"`
+}
+
+// ReaderDisplayContent represents the shape of the serialized protobuf sent to Rabbit Service for updating the reader display
+type ReaderDisplayContent struct {
+	Type               string             `json:"type"`
+	Cart               Cart               `json:"cart"`
+	TransactionContext TransactionContext `json:"transaction_context"`
+}
+
+// ReaderDisplayClearContent represents the shape of the serialized protobuf sent to Rabbit Service for clearing the reader display
+type ReaderDisplayClearContent struct {
+	TransactionContext TransactionContext `json:"transaction_context"`
+}
+
+// ChargeAmount belongs to the ReaderCollectPaymentContent protobuf send to Rabbit Service for collecting a specific payment after the Payment Intent is created
+type ChargeAmount struct {
+	ChargeAmount   int    `json:"charge_amount"`
+	Currency       string `json:"currency"`
+	TipAmount      int    `json:"tip_amount"`
+	CashbackAmount int    `json:"cashback_amount"`
+}
+
+// ReaderCollectPaymentContent represents the shape of the serialized protobuf sent to Rabbit Service for collecting a payment for a specific Payment Intent
+type ReaderCollectPaymentContent struct {
+	ChargeAmount       ChargeAmount       `json:"charge_amount"`
+	TransactionContext TransactionContext `json:"transaction_context"`
+}
+
+// ReaderQueryPaymentContent represents the shape of the serialized protobuf sent to Rabbit Service for querying a payment for its collection state
+type ReaderQueryPaymentContent struct {
+	TransactionContext TransactionContext `json:"transaction_context"`
+}
+
+// ReaderQueryPaymentResponse is the decoded RPC response from calling the queryPaymentMethod reader method
+type ReaderQueryPaymentResponse struct {
+	PaymentMethod interface{} `json:"payment_method"`
```

Comment:
For these `PaymentMethod`s, is there data you're pulling out of later? Wondering why they're `interface`s instead of [partially-defined] structs.

Reply:
we're not pulling any information out as the payment method will always be empty for a test payment. I wasn't able to get Go to encode/decode it to an actual empty object instead of `nil` unless I did it this way. A subsequent call to Stripe needs that to be an empty object in the request or it'll reject the payload.